src/edu/csus/ecs/pc2/core/util/IJSONTool.java

```diff
@@ -0,0 +1,285 @@
+     *
+     * @param account - the account to use to get the organization information
+     * @return Json object for the organization
+     */
+    public ObjectNode convertOrganizationsToJSON(Account account);
+
+    /**
+     * This converts an account to a team json object.
+     *
+     * @param account - the account
+     * @return Json object for the team
+     */
+    public ObjectNode convertToJSON(Account account);
+
+    /**
+     * This converts an account to a accounts json object.
+     *
+     * @param account - the account
+     * @return Json object for the account
+     */
+    public ObjectNode convertToAccountJSON(Account account);
+
+    /**
+     * This converts a problem to a json object.
+     *
+     * @param problem - the problem
+     * @param ordinal - the problem index
+     * @return Json object for the problem
+     */
+    public ObjectNode convertToJSON(Problem problem, int ordinal);
+
+    /**
+     * This routine looks up a specific Run Test Case from the array supplied and
+     * returns to a json object for that run test case.
+     *
+     * @param runTestCases - all the run test cases
+     * @param ordinal - index of desired run test case
+     * @return Json object for the run test case
+     */
+    public ObjectNode convertToJSON(RunTestCase[] runTestCases, int ordinal);
+
+    /**
+     * Create JSON for judgment.  If the submission hasn't been judged, that's fine,
+     * a boilerplate judgement is returned.
+     *
+     * @param submission/Run to fetch judgment from
+     * @return Json object for the judgement
+     */
+    public ObjectNode convertJudgementToJSON(Run submission);
+
+    /**
+     * Returns the unique group id for a group.  This could be the assigned group id, or
+     * the elementid if no group id is assigned.
+     *
+     * @param group
+     * @return string representation of the group id
+     */
+    public static String getGroupId(Group group) {
+        String id = group.getElementId().toString();
+        if (group.getGroupId() != -1) {
+            id = Integer.toString(group.getGroupId());
+        }
+        return id;
+    }
+
+    /**
+     * Returns organization (institution) number as a string
+     * @param account to use to fetch organization (institution) from
+     * @return string representation of the organization ID (which is a number)
+     */
+    public static String getOrganizationId(Account account) {
+        String id = account.getInstitutionCode();
+        if (id.startsWith("INST-U-")) {
+            id = id.substring(7);
+        }
+        if (id.startsWith("INST-")) {
+            id = id.substring(5);
+        }
+        return id;
+    }
+
+    /**
+     * Returns account ID as a string
+     * @param account to use to get the client id from
+     * @return string representation of the ID (which is a number)
+     */
+    public static String getAccountId(Account account) {
+        String id = account.getClientId().getName();
```

Comment:
It appears this line was changed.  Previously it read
```java
String id = "" + account.getClientId().getClientNumber();
```
Now it reads 
```java
String id = account.getClientId().getName();
```
Method `ClientId.getName()` returns a concatenation of the client type (e.g. `team`) with the client number (e.g. `3`).  However, the JavaDoc in this method's header says it is supposed to return a _number_ -- which isn't what `id` will contain after the call to `ClientId.getName()`.  Something's amiss somewhere...

Reply:
See the comment in the commit: "Need unique ID for event feed notification id for accounts, so can't use the "client number', must use the 'client name', eg. team3, administrator2, etc."    When I added "accounts" to the EF, several accounts had the same notification "id", eg adminsitrator1, team1, judge1, feeder1, scoreboard1 all had an id of "1".  Not acceptable.  This routine is only used by the JSON event feeder so (eg IJSONTool.java), so change the code in one place to address that problem.